### ISSUE TITLE
require(vCONST) broken

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14608]
Migrated from [rt.perl.org#124135](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=124135) (status was 'resolved')

Searchable as RT124135$

**Comment by p5pRT** [comment]
### From zefram@fysh.org

#### Created by zefram@fysh.org



$ perl5\.21\.4 \-E 'require\(v5\.6\.0\); say "OK"'
Can't locate  in @&#8203;INC \(@&#8203;INC contains&#8203;: /home/zefram/usr/perl/perl\_install/perl\-5\.21\.4\-i64\-f52/lib/site\_perl/5\.21\.4/x86\_64\-linux\-thread\-multi /home/zefram/usr/perl/perl\_install/perl\-5\.21\.4\-i64\-f52/lib/site\_perl/5\.21\.4 /home/zefram/usr/perl/perl\_install/perl\-5\.21\.4\-i64\-f52/lib/5\.21\.4/x86\_64\-linux\-thread\-multi /home/zefram/usr/perl/perl\_install/perl\-5\.21\.4\-i64\-f52/lib/5\.21\.4 \.\) at \-e line 1\.
$ perl5\.21\.3 \-E 'require\(v5\.6\.0\); say "OK"'
OK

Still broken in 5\.21\.10\.  Also affects 'require\(5\.6\.0\)'\.
Doesn't affect '$a=v5\.6\.0; require\($a\)'\.  I suspect commit
ed6f4477d0922cc828d8bc6fd16e08c01f7c1210\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.21.4:

Configured by zefram at Fri Mar 20 21:01:29 GMT 2015.

Summary of my perl5 (revision 5 version 21 subversion 4) configuration:
   
  Platform:
    osname=linux, osvers=3.2.0-4-amd64, archname=x86_64-linux-thread-multi
    uname='linux barba.rous.org 3.2.0-4-amd64 #1 smp debian 3.2.65-1+deb7u1 x86_64 gnulinux '
    config_args='-des -Dprefix=/home/zefram/usr/perl/perl_install/perl-5.21.4-i64-f52 -Duselargefiles -Dusethreads -Uafs -Ud_csh -Uusesfio -Uusenm -Duseshrplib -Dusedevel -Uversiononly -Ui_db'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=2',
    optimize='-O2',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fwrapv -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.7.2', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16, longdblkind=3
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -fstack-protector -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/gcc/x86_64-linux-gnu/4.7/include-fixed /usr/include/x86_64-linux-gnu /usr/lib /lib/x86_64-linux-gnu /lib/../lib /usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib
    libs=-lnsl -ldb -ldl -lm -lcrypt -lutil -lpthread -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=libc-2.19.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.19'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E -Wl,-rpath,/home/zefram/usr/perl/perl_install/perl-5.21.4-i64-f52/lib/5.21.4/x86_64-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector'



@INC for perl 5.21.4:
    /home/zefram/usr/perl/perl_install/perl-5.21.4-i64-f52/lib/site_perl/5.21.4/x86_64-linux-thread-multi
    /home/zefram/usr/perl/perl_install/perl-5.21.4-i64-f52/lib/site_perl/5.21.4
    /home/zefram/usr/perl/perl_install/perl-5.21.4-i64-f52/lib/5.21.4/x86_64-linux-thread-multi
    /home/zefram/usr/perl/perl_install/perl-5.21.4-i64-f52/lib/5.21.4
    .


Environment for perl 5.21.4:
    HOME=/home/zefram
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/zefram/usr/perl/perl_install/perl-5.21.4-i64-f52/bin:/home/zefram/usr/perl/util:/home/zefram/pub/x86_64-unknown-linux-gnu/bin:/home/zefram/pub/common/bin:/usr/bin:/bin:/usr/local/bin:/usr/games
    PERL_BADLANG (unset)
    SHELL=/usr/bin/zsh

```
</details>

**Comment by p5pRT** [comment]
### From @wolfsage

On Fri\, Mar 20\, 2015 at 5&#8203;:48 PM\, Zefram \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> Still broken in 5\.21\.10\.  Also affects 'require\(5\.6\.0\)'\.
> Doesn't affect '$a=v5\.6\.0; require\($a\)'\.  I suspect commit
> ed6f4477d0922cc828d8bc6fd16e08c01f7c1210\.

A bisect lead me to this commit as well\.

\-\- Matthew Horsfall \(alh\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @jkeenan

On Fri Mar 20 14&#8203;:55&#8203;:12 2015\, alh wrote&#8203;:
> On Fri\, Mar 20\, 2015 at 5&#8203;:48 PM\, Zefram \<perlbug\-followup@&#8203;perl\.org> wrote&#8203;:
> > Still broken in 5\.21\.10\.  Also affects 'require\(5\.6\.0\)'\.
> > Doesn't affect '$a=v5\.6\.0; require\($a\)'\.  I suspect commit
> > ed6f4477d0922cc828d8bc6fd16e08c01f7c1210\.
> 
> A bisect lead me to this commit as well\.
> 
> \-\- Matthew Horsfall \(alh\)
> 

Grrrr\.  That commit changed behavior in a fundamental file \(op\.c\) but did not include a test\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
### From @cpansprout

On Fri Mar 20 16&#8203;:26&#8203;:51 2015\, jkeenan wrote&#8203;:
> On Fri Mar 20 14&#8203;:55&#8203;:12 2015\, alh wrote&#8203;:
> > On Fri\, Mar 20\, 2015 at 5&#8203;:48 PM\, Zefram \<perlbug\-followup@&#8203;perl\.org>
> > wrote&#8203;:
> > > Still broken in 5\.21\.10\.  Also affects 'require\(5\.6\.0\)'\.
> > > Doesn't affect '$a=v5\.6\.0; require\($a\)'\.  I suspect commit
> > > ed6f4477d0922cc828d8bc6fd16e08c01f7c1210\.
> >
> > A bisect lead me to this commit as well\.
> >
> > \-\- Matthew Horsfall \(alh\)
> >
> 
> Grrrr\.  That commit changed behavior in a fundamental file \(op\.c\) but
> did not include a test\.

It was not supposed to change the behaviour at all\, but simply speed things up\.

In any case\, it is now fixed in 2b79cfef751\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'
